### DISCUSSION TITLE
How to get value from created item via MultiSelect?

### DISCUSSION THREAD

**Discussion by ehsanonline** [mantinedev/mantine#1473] [Feature requests]
I'm using `MultiSelect` with data which have values and labels, the problem is when I use `OnCreate` to create new Item, instead of value, the label will pass to `OnChange` and will add to the list. the `data` property works fine and new Item with value is there, but not in the OnChange. How can I pass the whole new Item into OnChange? 
Thanks.

**Comment by rtivital** [comment]
`onCreate` function is called with a value that user entered, it's up to you to assign a label to it.

**Reply by ehsanonline** [reply]
I get it now, it should be the value that user enters, not the label. I think it's a better UX if user enters a label, then a [generated] value returned from `OnCreate`, and passed into `OnChange` along with other selected items.

**Comment by ehsanonline** [comment]
This is not about `label`, it's about value in data list, perhaps an example makes it more clear:
```js
const [data,setData] = useState([
  {value:'1', label:'an item'},
  {value:'6', label:'an other item'},
])
 <MultiSelect
  data={data}
  creatable
  getCreateLabel={(query) => `+ ${query}`}
  onCreate={(query) => setData([...data,  {value:'10', label:'new item'}  ])}
  onChange={(value) => {
    console.log(value) // ["new item"] will be printed, But I need the value from item : {value:'10', label:'new item'}
  }}
/>
```

**Reply by rtivital** [reply]
You need to use controlled component and set value on your side in `onCreate`

**Reply by ehsanonline** [reply]
I've tried that, the problem is when creating new item, the `onChange` will be triggred too. which has label (instead of value). and with onChange there is no way to detect either it came from a selected item (which has proper value) or a created item (which has just label).

**Reply by rtivital** [reply]
Okay, thanks for reporting, we'll have a look, @achmurali can you check that?
